src/client/Client.java

```diff
@@ -101,30 +98,42 @@ public boolean login(String username, String password) throws IOException, Class
                 null,
                 Message.Type.LOGIN,
                 Message.Status.REQUEST,
-                String.format("username: %s password: %s", username, password));
+                String.format("username: %s password: %s", username, password),
+                -1);
 
         write.writeObject(m);
         Message res = (Message) read.readObject();
-        return res.getType() == Message.Type.LOGIN && res.getStatus() == Message.Status.SUCCESS;
+        if (res.getType() == Message.Type.LOGIN && res.getStatus() == Message.Status.SUCCESS) {
+            user.setUserId(res.getSenderId());
+            user.setConversationId(res.getConversationId());
+            return true;
+        } else {
+            return false;
+        }
     }
 
     public boolean logout() throws IOException, ClassNotFoundException {
-        boolean success = false;
-        Message response;
         Message m = new Message(
                 user.getUserId(),
                 null,
                 Message.Type.LOGOUT,
                 Message.Status.REQUEST,
-                "Logging out!");
+                "Logging out!",
+                user.getConversationId());
```

Comment:
Logout/login messages are special system messages, they probably don't need a conversation ID. Could cause conversations to get flooded w/ login logout messages if not handled carefully.